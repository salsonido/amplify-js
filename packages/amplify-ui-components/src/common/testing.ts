import {
	boolean as booleanKnob,
	select as selectKnob,
	text as textKnob,
} from '@storybook/addon-knobs';

export const pixelThreshold = 10;

export const selectors = {
	// amplify-form-field
	formFieldDescription: '[data-test=form-field-description]',
};

export const knobs = {
	labelKnob(labelPlaceholder: string) {
		return textKnob('Label', labelPlaceholder);
	},

	descriptionKnob(descriptionPlaceholder: string) {
		return textKnob('Description', descriptionPlaceholder);
	},

	hintKnob(hintPlaceholder: string) {
		return textKnob('Hint', hintPlaceholder);
	},

	inputTypeKnob(inputTypePlaceholder: string) {
		return selectKnob(
			'Type',
			[
				'date',
				'email',
				'number',
				'password',
				'search',
				'tel',
				'text',
				'url',
				'time',
			],
			inputTypePlaceholder
		);
	},

	placeholderKnob(placeholderPlaceholder: string) {
		return textKnob('Placeholder', placeholderPlaceholder);
	},

	buttonTypeKnob(buttonTypePlaceholder: string) {
		return selectKnob(
			'Button type',
			['button', 'submit', 'reset'],
			buttonTypePlaceholder
		);
	},

	buttonTextKnob(buttonTextPlaceholder: string) {
		return textKnob('Button text', buttonTextPlaceholder);
	},

	toggleKnob(toggleDefault: boolean) {
		return booleanKnob('Toggle', toggleDefault);
	},

	disabledKnob(disabledDefault: boolean) {
		return booleanKnob('Disabled', disabledDefault);
	},

	formSectionHeaderKnob(headerPlaceholder: string) {
		return textKnob('Header text', headerPlaceholder);
	},

	formSectionSubmitKnob(submitPlaceholder: string) {
		return textKnob('Submit button text', submitPlaceholder);
	},

	tooltipAutoShowKnob(autoShowDefault: boolean) {
		return booleanKnob('Tooltip autoshow', autoShowDefault);
	},
};
