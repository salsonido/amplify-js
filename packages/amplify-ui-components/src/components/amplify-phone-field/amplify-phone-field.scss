.phone-field {
	display: inline-flex;
	align-items: flex-start;
	width: 100%;
	input {
		border-left: none;
		border-radius: 0 3px 3px 0;
	}
}
