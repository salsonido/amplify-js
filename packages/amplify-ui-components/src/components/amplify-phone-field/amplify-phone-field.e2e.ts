import { newE2EPage } from '@stencil/core/testing';
import { pixelThreshold } from '../../common/testing';

describe('amplify-phone-field e2e:', () => {
	it('should render a amplify-phone-field', async () => {
		const page = await newE2EPage();
		await page.setContent(`<amplify-phone-field></amplify-phone-field>`);
		const el = await page.find('amplify-phone-field');
		expect(el).not.toBeNull();

		const screenshot = await page.compareScreenshot('amplify-phone-field', {
			fullPage: true,
		});
		expect(screenshot).toMatchScreenshot({
			allowableMismatchedPixels: pixelThreshold,
		});
	});
});
