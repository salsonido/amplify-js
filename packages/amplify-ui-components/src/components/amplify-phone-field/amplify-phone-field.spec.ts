import { I18n } from '@aws-amplify/core';
import { newSpecPage } from '@stencil/core/testing';
import { AmplifyPhoneField } from './amplify-phone-field';
import { PHONE_SUFFIX } from '../../common/constants';
import { Translations } from '../../common/Translations';

describe('amplify-phone-field spec:', () => {
	describe('Component logic ->', () => {
		let phoneField;

		beforeEach(() => {
			phoneField = new AmplifyPhoneField();
		});

		it('should render `phone` for the field id', () => {
			expect(phoneField.fieldId).toEqual(PHONE_SUFFIX);
		});

		it('should have `label` render to `Verification phone` by default', () => {
			expect(phoneField.label).toEqual(I18n.get(Translations.PHONE_LABEL));
		});

		it('should have `placeholder` render to `Enter phone` by default', () => {
			expect(phoneField.placeholder).toEqual(
				I18n.get(Translations.PHONE_PLACEHOLDER)
			);
		});

		it('should have `required` set to `false` by default', () => {
			expect(phoneField.required).toBe(false);
		});
	});
	describe('Render logic ->', () => {
		it('should render `placeholder` and `label values', async () => {
			const page = await newSpecPage({
				components: [AmplifyPhoneField],
				html: `<amplify-phone-field label="New phone *" placeholder="Enter new phone"></amplify-phone-field>`,
			});

			expect(page.root).toMatchSnapshot();
		});

		it('should render default values by default', async () => {
			const page = await newSpecPage({
				components: [AmplifyPhoneField],
				html: `<amplify-phone-field></amplify-phone-field>`,
			});

			expect(page.root).toMatchSnapshot();
		});
	});
});
