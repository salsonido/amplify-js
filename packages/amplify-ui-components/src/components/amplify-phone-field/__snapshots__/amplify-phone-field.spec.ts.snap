// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-phone-field spec: Render logic -> should render \`placeholder\` and \`label values 1`] = `
<amplify-phone-field label="New phone *" placeholder="Enter new phone">
  <div>
    <amplify-form-field label="New phone *">
      <div class="phone-field" slot="input">
        <amplify-country-dial-code></amplify-country-dial-code>
        <amplify-input fieldid="phone" name="phone" placeholder="Enter new phone" type="tel"></amplify-input>
      </div>
    </amplify-form-field>
  </div>
</amplify-phone-field>
`;

exports[`amplify-phone-field spec: Render logic -> should render default values by default 1`] = `
<amplify-phone-field>
  <div>
    <amplify-form-field label="Phone Number *">
      <div class="phone-field" slot="input">
        <amplify-country-dial-code></amplify-country-dial-code>
        <amplify-input fieldid="phone" name="phone" placeholder="(555) 555-1212" type="tel"></amplify-input>
      </div>
    </amplify-form-field>
  </div>
</amplify-phone-field>
`;
